app/utils/roadmap-sorting.ts

```diff
@@ -0,0 +1,53 @@
+import seedrandom from 'seedrandom';
+
+export function getRoughWeekSeed(date: number): number {
+  const now = date;
+  const weekMs = 7 * 24 * 60 * 60 * 1000; // milliseconds per week
+
+  return now - (now % weekMs);
+}
+
+export function getSortPositionForRoadmapPage(
+  developmentStatus: string,
+  votesCount: number,
+  id: string,
+  date: number,
+  currentUserId?: string | null,
+): string {
+  const statusPriority =
+    {
+      in_progress: 1,
+      not_started: 2,
+      released: 3,
+    }[developmentStatus] ?? 4; // Default to lowest priority if status is unknown
+
+  const sortKeys: string[] = [statusPriority.toString()];
+
+  if (currentUserId) {
+    // For authenticated users: fixed order per user per week
+    const weekSeed = getRoughWeekSeed(date);
+    const userWeekSeed = `${currentUserId}-${weekSeed}`;
```

Comment:
It this is only used in the second branch (`not_started`), instantiate it there instead - placing it outside the branches makes it look like this is common to 1+ branches.